src/explib/datasets.py

```diff
@@ -30,6 +32,13 @@ def __init__(
         else:
             self.dataset = pd.read_csv(dataset).values
 
+        #
+        self.dataset = self.dataset.to(device)
+        if not isinstance(labels, torch.Tensor):
```

Comment:
If dataset is a torch.utils.data.Dataset we don't need labels right? It automatically has a getitem method that returns the data and the label. So I would need to add a check here.

Reply:
I think The baseclass __getitem__ method is going to fail if the dataset is only a np array or a torch.Tensor. I understand there is the correct implementation in the derived class but maybe here we would need to put a check or raise an error?